From d25e6c87cd0d1c33206018d94d62aa487fc790c8 Mon Sep 17 00:00:00 2001
From: Jackie Huang <jackie.huang@windriver.com>
Date: Thu, 27 May 2021 15:42:48 +0800
Subject: [PATCH 10/12] kubernetes.pp: add re-tries for docker login

The docker login to local registry may sometimes fail:
Error: docker login registry.local:9001 -u admin -p St8rlingX* returned
1 instead of one of [0]

It's rare and the root cause is not found yet, but a re-try will always
succeed,
so add re-tries and a try sleep between re-tries to workaround it for
now.

Upstream-Status: Inappropriate [workaround]

Rebased for stx 5.0

Signed-off-by: Jackie Huang <jackie.huang@windriver.com>
---
 puppet-manifests/src/modules/platform/manifests/kubernetes.pp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/puppet-manifests/src/modules/platform/manifests/kubernetes.pp b/puppet-manifests/src/modules/platform/manifests/kubernetes.pp
index e7e9c51..235618c 100644
--- a/puppet-manifests/src/modules/platform/manifests/kubernetes.pp
+++ b/puppet-manifests/src/modules/platform/manifests/kubernetes.pp
@@ -279,6 +279,8 @@ class platform::kubernetes::master::init
     -> exec { 'remove taint from master node':
       command   => "kubectl --kubeconfig=/etc/kubernetes/admin.conf taint node ${::platform::params::hostname} node-role.kubernetes.io/master- || true", # lint:ignore:140chars
       logoutput => true,
+      tries     => 3,
+      try_sleep => 1,
     }
 
     # Add kubelet service override
@@ -497,6 +499,8 @@ class platform::kubernetes::coredns {
       exec { 'restrict coredns to master nodes':
         command   => 'kubectl --kubeconfig=/etc/kubernetes/admin.conf -n kube-system patch deployment coredns -p \'{"spec":{"template":{"spec":{"nodeSelector":{"node-role.kubernetes.io/master":""}}}}}\'', # lint:ignore:140chars
         logoutput => true,
+        tries     => 3,
+        try_sleep => 1,
       }
 
       -> exec { 'Use anti-affinity for coredns pods':
-- 
2.7.4

